### PR TITLE
Fix compilation on nightly

### PR DESCRIPTION
**Pull request by konstin**
rust-lang/rust#50120 has made two breaking changes that break pyo3's compilation:

 * Custom attributes must only have a path of length one, longer paths got an extra feature gate `proc_macro_path_invoc`. This means that e.g. `#[py::modinit(rust2py)]` fails to compile. 
 * The attr TokenStream no longer gets the braces, e.g. it now only gets `rust2py` and not `(rust2py)`.

This pull request fixes both. I've tried to avoid introducing breaking changes and instead replaced `py::modinit` with a `pymodinit` alias. I wouldn't mind removing the alias or renaming the proc macros.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -83,10 +83,12 @@ features = ["extension-module"]
 extern crate pyo3;
 use pyo3::prelude::*;
 
+use pyo3::py::modinit as pymodinit;
+
 // add bindings to the generated python module
 // N.B: names: "librust2py" must be the name of the `.so` or `.pyd` file
 /// This module is implemented in Rust.
-#[py::modinit(rust2py)]
+#[pymodinit(rust2py)]
```
Comment (fafhrd91) [inline README.md]: What is the reason for this change?

Comment (konstin) [reply README.md]: `py::modinit` doesn't work without the `proc_macro_path_invoc` feature, which isn't expected to be stabilized soon (unlike proc macros and specialization).

Comment (fafhrd91) [reply README.md]: I see

**Comment by fafhrd91** [comment]
you should have permission to merge

**Comment by konstin** [comment]
Yes, thanks!

**Comment by konstin** [comment]
@fafhrd91 Do you have any policy on when to make pull requests and when to directly push to master?

**Comment by fafhrd91** [comment]
I don't really have any policy, but master branch is protected :)
I prefer for someone to review PR before merge

**Comment by fafhrd91** [comment]
@konstin @althonos I think you should be able to review PRs
